Gems/EMotionFX/Code/EMotionFX/Source/AnimGraphNodeGroup.h

```diff
@@ -174,13 +174,26 @@ namespace EMotionFX
          */
         bool IsNameEditOngoing() const;
 
+        /**
+         * Returns a the id of the parent AnimGraphNode to which this group belongs.
+         * If the group belongs to the root level, this function will return an invalid id.
+         * This function is used to make sure that groups can be filtered by level.
+         */
+        AnimGraphNodeId ParentNodeId() const;
+
+        /**
+         * Sets the id of the parent AnimGraphNode for this group.
+         */
+        void SetParentNodeId(AnimGraphNodeId nodeId);
+
         static void Reflect(AZ::ReflectContext* context);
 
     protected:
         AZStd::vector<AZ::u64>  m_nodeIds;          /**< The node ids that are inside this group. */
         AZStd::string           m_name;             /**< The unique identification number for the node group name. */
         AZ::u32                 m_color;            /**< The color the nodes of the group will be filled with. */
         bool                    m_isVisible;
-        bool m_nameEditOngoing = false; /**< Whether the user is currently typing a new name */
+        bool                    m_nameEditOngoing = false; /**< Whether the user is currently typing a new name */
+        AnimGraphNodeId         m_parentNodeId; /**< The id of the parent AnimGraphNode */
```

Comment:
question: Just curious does `AnimGraphNodeId` get default constructed to invalid? Or if not would it be good to assign it to an invalid value here as a member initializer?

Reply:
`AnimGraphNodeId` is an alias for `ObjectId` which gets default constructed to `Invalid`

Reply:
Perfect thanks for checking!